src/components/UserPage/UserPage.js

```diff
@@ -1,10 +1,135 @@
-import React, { Component } from 'react';
+import React, { Component, useEffect } from 'react';
 import { connect } from 'react-redux';
 import LogOutButton from '../LogOutButton/LogOutButton';
 import mapStoreToProps from '../../redux/mapStoreToProps';
 
+const mapStateToProps = (state, ownProps) => ({
+  dispatch: ownProps.dispatch,
+  history: ownProps.history,
+  user: state.user,
+  friends: state.friendsReducer,
+  decks: state.deck,
+})
+
+const UserPageFunc = ({
+  dispatch,
+  history,
+  user,
+  friends,
+  decks,
+}) => {
+  useEffect(() => {
+    dispatch({ type: 'FETCH_USER' });
+    dispatch({ type: 'GET_DECK' });
+    dispatch({ type: 'GET_FRIENDS' });
+  }, [dispatch])
+
+  const createDeck = () => {
+    const deck = prompt("Please enter a deck name:", "My Awesome Deck");
+    if (deck === null || deck === "") {
+      alert("Canceled");
+    } else {
+      this.props.dispatch({
+        type: 'CREATE_DECK',
+        payload: {
+          userid: user.id,
+          deckname: deck,
+          ispublic: false,
+          description: "",
+          decklist: '',
+          featured_card: '',
+          upvotes: 0,
+          comments: '',
+        },
+      });
+    };
+  };
```

Comment:
I cleaned up this method. In functional components you have to declare them as 'var/const/let'. The good part of that is you no longer need the 'this.' keyword everywhere.
In the first if statement, your code has a return. If you are using 'else' you don't need to bail in the first if block.
Since this is a create there are only two parts of the payload that are influenced by the user. Their id and the deckname (also should be deckName). The rest of the object field can be applied when you are setting this at the other end of this action (the reducer). Also 'featured_card', should be 'featuredCard'. '_' are usually reserved for data side fields. On the front end we want to camel case.